api/src/repositories/security-repository.ts

```diff
@@ -0,0 +1,124 @@
+  persecution_or_harm_id: z.number(),
+  persecution_or_harm_type_id: z.number(),
+  wldtaxonomic_units_id: z.number(),
+  name: z.string(),
+  description: z.string().nullable().optional()
+});
+
+export type PersecutionAndHarmSecurity = z.infer<typeof PersecutionAndHarmSecurity>;
+
+export const SecurityReason = z.object({
+  id: z.number(),
+  type_id: z.number()
+});
+
+export type SecurityReason = z.infer<typeof SecurityReason>;
+
+/**
+ * A repository for maintaining security on artifacts.
+ *
+ * @export
+ * @class SecurityRepository
+ * @extends BaseRepository
+ */
+export class SecurityRepository extends BaseRepository {
+  /**
+   * Get persecution and harm rules.
+   *
+   * @return {*}  {Promise<PersecutionAndHarmSecurity[]>}
+   * @memberof SecurityRepository
+   */
+  async getPersecutionAndHarmRules(): Promise<PersecutionAndHarmSecurity[]> {
+    defaultLog.debug({ label: 'getPersecutionAndHarmRules' });
+
+    const sqlStatement = SQL`
+      SELECT
+        persecution_or_harm_id,
+        persecution_or_harm_type_id,
+        wldtaxonomic_units_id,
+        name,
+        description
+      FROM
+        persecution_or_harm;
+    `;
+
+    const response = await this.connection.sql<PersecutionAndHarmSecurity>(sqlStatement, PersecutionAndHarmSecurity);
+
+    const results = (response.rowCount && response.rows) || null;
+
+    if (!results) {
+      throw new ApiExecuteSQLError('Failed to get persecution and harm rules');
+    }
+
+    return results;
+  }
+
+  /**
+   * Apply security rules to an artifact.
+   *
+   * @param {number} artifactId
+   * @param {number} securityId
+   * @return {*}  {Promise<{ artifact_persecution_id: number }>}
+   * @memberof SecurityRepository
+   */
+  async applySecurityRulesToArtifact(
+    artifactId: number,
+    securityId: number
+  ): Promise<{ artifact_persecution_id: number }> {
+    defaultLog.debug({ label: 'applySecurityRulesToArtifact' });
+
+    const sqlStatement = SQL`
+      INSERT INTO artifact_persecution (
+        artifact_id,
+        persecution_or_harm_id
+      ) VALUES (
+        ${artifactId},
+        ${securityId}
+      )
+      RETURNING artifact_persecution_id;
+    `;
+
+    const response = await this.connection.sql<{ artifact_persecution_id: number }>(sqlStatement);
+
+    const results = (response.rowCount && response.rows[0]) || null;
+
+    if (!results) {
+      throw new ApiExecuteSQLError('Failed to apply security rules to artifact');
+    }
+
+    return results;
+  }
+
+  /**
+   * Remove a security rule from an artifact.
+   *
+   * @param {number} artifactId
+   * @return {*}  {Promise<void>}
+   * @memberof SecurityRepository
+   */
+  async removeAllSecurityRulesFromArtifact(artifactId: number): Promise<void> {
+    defaultLog.debug({ label: 'removeAllSecurityRulesFromArtifact' });
+
+    const sqlStatement = SQL`
+      DELETE FROM
+        artifact_persecution
+      WHERE
+        artifact_id = ${artifactId};
+    `;
```

Comment:
Can do this, but my consideration was that on the endpoint or frontend, we are doing nothing with that return. I could return and run a check if something were to return. But would just create more testing cases, if there were to be an erorr that happens in the DB then it will still be caught by endpoint try/catch

Reply:
Yeah totally fair. Feel free to ignore